src/less/button/button.less

```diff
@@ -354,7 +354,10 @@ a.fake-btn--form {
     border-color: inherit;
     .border-radius-token(expand-btn-border-radius, border-radius-50);
     max-width: 100%;
+}
 
+button.btn--form:not([disabled], [aria-disabled="true"]),
```

Comment:
The changes here are not in scope with the issue/PR titles. You seem to be applying the fix to listbox button as well as all buttons and fake buttons. Maybe update the issue and PR title.